pkg/runtime/framework/plugins/torch/torch.go

```diff
@@ -49,9 +52,39 @@ func (t *Torch) Name() string {
 	return Name
 }
 
-// TODO: Need to implement validations for Torch policy.
-func (t *Torch) Validate(oldObj, newObj *trainer.TrainJob) (admission.Warnings, field.ErrorList) {
-	return nil, nil
+func (t *Torch) Validate(runtimeJobTemplate client.Object, runtimeInfo *runtime.Info, oldObj, newObj *trainer.TrainJob) (admission.Warnings, field.ErrorList) {
+	var allErrs field.ErrorList
+	if runtimeInfo == nil || runtimeInfo.RuntimePolicy.MLPolicy == nil || runtimeInfo.RuntimePolicy.MLPolicy.Torch == nil || newObj.Spec.Trainer.NumProcPerNode == nil {
+		return nil, allErrs
+	}
+
+	specPath := field.NewPath("spec")
+
+	if newObj.Spec.Trainer != nil {
+		numProcPerNodePath := specPath.Child("trainer").Child("numProcPerNode")
+		numProcPerNode := *newObj.Spec.Trainer.NumProcPerNode
+		if numProcPerNode.Type == intstr.String {
+			allowedStringValList := []string{"auto", "cpu", "gpu"}
```

Comment:
```suggestion
			allowed := sets.New("auto", "cpu", "gpu")
```

1. We want to avoid typed name in variable name
2. Sets allows us iteration rather than slice, especially this case